src/pages/Content/index.js

```diff
@@ -84,247 +42,311 @@ async function getUserInfo() {
+    tooltip.style.visibility = 'hidden';
+    tooltip.style.display = 'none';
+
+    const renderToolTip = (mouseX, mouseY, selection) => {
+      mouseX = mouseX - 50;
+      mouseY = mouseY - 40;
+      if (selection.length < 60) {
+        mouseX = mouseX - 20;
+      }
+      tooltip.style.top = mouseY + 'px';
+      tooltip.style.left = mouseX + 'px';
+      tooltip.style.visibility = 'visible';
+      tooltip.style.display = 'block';
+      isToolTipVisible = true;
+    };
+
+    const closeToolTip = () => {
+      tooltip.style.visibility = 'hidden';
+      tooltip.style.display = 'none';
+      isToolTipVisible = false;
+      clearSelection();
+    };
+
+    function clearSelection() {
+      if (window.getSelection) {window.getSelection().removeAllRanges();}
+      else if (document.selection) {document.selection.empty();}
+    }
+
+    var startX = 0;
+    var endX = 0;
+    var startY = 0;
+    var endY = 0;
+
+    //Close the tool tip
+    document.addEventListener('mousedown', (e) => {
+      const parentClassName = e.target.parentNode.getAttribute('class');
+      const parentIdName = e.target.parentNode.getAttribute('id');
+      console.log('Parent Class:', parentClassName);
+      console.log('Parent ID:', parentIdName);
+      /*
+        ISSUE 1: Need more testing on more pages to see if the tooltip properly disappears
+        You can double click to "remove" a selection, but the tooltip is still visible need to workout cases
+        
+        Albeit this one may be a very simple fix
+        Share findings if anyone can.
+      */
+
+      //Make the tool tip invisible
+      if (isToolTipVisible) {
+        e.stopPropagation();
+        return false;
+      } else {
+        startX = e.pageX;
+        startY = e.pageY;
+        closeToolTip();
+      }
+    });
+    // Show the tool tip
+    document.addEventListener('mouseup', (e) => {
+      console.log(window.getSelection());
+      let temp = window.getSelection();
+      let selection = temp.toString();
+      console.log('Current selection', selection);
+      console.log('THIS SELECTION WAS THIS LONG: ', selection.length);
+      console.log('Selection baseNode:', temp.baseNode);
+      console.log('Selection focusNode:', temp.focusNode);
+      
+      /*
+      ISSUE 2:
+      The comp variable is made as one of the checks for NOT displaying the toolbar.
+
+      In short,
+        1. If the baseNode and focusNode of the selection are equal then we can display toolbar
+        2. If the baseNode's parentNode is equal to the focusNode's parentNode then we can display the toolbar
+        3. If the focusNode's grandparent (if existing) is the the baseNode's parentNode then we can display the toolbar
+
+        I added #3 to allow links to be highlighted, but it doesn't always work for nested highlights (refer to ISSUE 4)
+
+        For more reference checkout: https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model
+      */
+       let comp =
```

Comment:
### ISSUE 2:
The comp variable is made as one of the checks for NOT displaying the toolbar.
In short,
1. If the baseNode and focusNode of the selection are equal then we can display toolbar
2. If the baseNode's parentNode is equal to the focusNode's parentNode then we can display the toolbar
3. If the focusNode's grandparent (if existing) is the the baseNode's parentNode then we can display the toolbar

I added number 3 to allow links to be highlighted, but it doesn't always work for nested highlights (refer to ISSUE 4)
For more reference checkout: https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model